deebotozmo/cli.py

```diff
@@ -0,0 +1,343 @@
+        print("Stats Type: " + str(event.type))
+        lock.release()
+    listener = vacbot.bot.statsEvents.subscribe(on_stats_event)
+    await vacbot.bot.execute_command(GetStats())
+    await lock.acquire()
+    vacbot.bot.statsEvents.unsubscribe(listener)
+    await vacbot.goodbye()
+
+# Needs a lock but the lifespan event is emitted
+# for every component, so it wouldn't work
+@cli.command(help='Get robot components life span')
+@coro
+async def components():
+    vacbot = dologin()
+    await vacbot.run()
+
+    async def on_lifespan_event(event: LifeSpanEvent):
+        print(str(event.type) + ": " + str(event.percent) + "%")
+    listener = vacbot.bot.lifespanEvents.subscribe(on_lifespan_event)
+    await vacbot.bot.execute_command(GetLifeSpan())
+    vacbot.bot.lifespanEvents.unsubscribe(listener)
+    await vacbot.goodbye()
+
+@cli.command(help='Get saved rooms')
+@coro
+async def getrooms():
+    print ("Not supported, no clue how to make it work.")
+    return
+    dologin()
+    vacbot.refresh_statuses()
+
+    for v in vacbot.getSavedRooms():
+        print(str(v['id']) + ' ' + v['subtype'])
+
+@cli.command(help='Get robot map and save it [filepath ex: "/folder/livemap.png"')
+@click.argument('filepath', type=click.STRING, required=True)
+@coro
+async def exportLiveMap(filepath):
+    vacbot = dologin()
+    await vacbot.run()
+
+    lock = asyncio.Lock()
+    await lock.acquire()
+    async def on_map(event: MapEvent):
+        with open(filepath, "wb") as fh:
+            fh.write(base64.decodebytes(vacbot.bot.map.get_base64_map()))
+        lock.release()
+    listener = vacbot.bot.map.mapEvents.subscribe(on_map)
+    await vacbot.bot.execute_command(GetMapTrace())
+    await vacbot.bot.execute_command(GetPos())
+    await vacbot.bot.execute_command(GetMajorMap())
+    await lock.acquire()
+    vacbot.bot.map.mapEvents.unsubscribe(listener)
+
+    await vacbot.goodbye()
+
+class dologin:
+    def __init__(self):
+        return
+
+    async def run(self):
+        if not config_file_exists():
+            click.echo("Not logged in. Do '%s createconfig' first." % (os.path.basename(sys.argv[0]),))
+            exit(1)
+
+        config = read_config()
+
+        self.session = aiohttp.ClientSession()
+
+        self.api = EcovacsAPI(self.session, config['device_id'], config['email'], config['password_hash'],
+                                    continent=config['continent'], country=config['country'],
+                                    verify_ssl=bool(config['verify_ssl']))
+        await self.api.login()
+
+        self.devices_ = await self.api.get_devices()
+
+        self.auth = await self.api.get_request_auth()
+        self.bot = VacuumBot(self.session, self.auth, self.devices_[0], continent=config['continent'],
+                                   country=config['country'], verify_ssl=bool(config['verify_ssl']))
+
+        self.mqtt = EcovacsMqtt(self.auth, continent=config['continent'], country=config['country'])
```

Comment:
```suggestion
```
You don't need it as mqtt is used to get events, which are triggered from other instances. As you unscribe straight away you would never get these events. I don't know how you can run mqtt in the background and inform you about new events